core/src/main/java/software/amazon/awssdk/http/pipeline/stages/ApplyUserAgentStage.java

```diff
@@ -31,20 +32,38 @@
 public class ApplyUserAgentStage implements MutableRequestToRequestPipeline {
 
     private final LegacyClientConfiguration config;
+    private final String clientName;
+    private final String clientType;
 
-    public ApplyUserAgentStage(HttpClientDependencies dependencies) {
+    public ApplyUserAgentStage(HttpClientDependencies dependencies,
+                               String clientName,
+                               String clientType) {
```

Comment:
I would forgo the type since that should be obvious from the client name. You can also get the client name from the dependencies so you don't need to do a custom supplier.